tests/javascript/core/spec.js

```diff
@@ -9,180 +9,191 @@
+		});
+	});
+
+	describe('Core Joomla.checkAll', function () {
+		var form = document.getElementById('check-all-form');
+		form.boxchecked = {};
+		var element = document.getElementById('cb0');
+
+		beforeAll(function () {
+			Joomla.checkAll(element);
+		});
+
+		it('should return false when input element is not inside a form', function () {
+			expect(Joomla.checkAll(document.getElementById('cb-no-form'))).toEqual(false);
+		});
+
+		it('should check all the checkboxes that has id starting with \'cb\' inside the form', function () {
+			expect($('#cb0')).toBeChecked();
+			expect($('#cb1')).toBeChecked();
+			expect($('#cb2')).toBeChecked();
+			expect($('#no-cb3')).not.toBeChecked();
+		});
+
+		it('should set the number of checked boxes in the form to form.boxchecked', function () {
+			expect(form.boxchecked.value).toEqual(3);
+		});
+
+		it('should use passed in stub to look for input elements', function () {
+			var element = document.getElementById('stub-check-test-1');
+			Joomla.checkAll(element, 'stub');
+
+			expect($('#stub-check-test-1')).toBeChecked();
+			expect($('#stub-check-test-2')).toBeChecked();
+		});
+	});
+
+	describe('Core Joomla.renderMessages and Joomla.removeMessages', function () {
+		var messages = {
+			"message": ["Message one", "Message two"],
+			"error": ["Error one", "Error two"]
+		};
+
+		beforeAll(function () {
+			Joomla.JText.load({"message": "Message"});
+			Joomla.renderMessages(messages);
+		});
+
+		it('renderMessages should render titles when translated strings are available', function () {
+			expect($('h4.alert-heading').first()).toContainText('Message');
+		});
+
+		it('renderMessages should render messages inside a div having class alert-message', function () {
+			var $messages = $('div.alert-success').children('div');
+			expect($messages[0]).toContainText('Message two');
+			expect($messages[1]).toContainText('Message one');
+		});
+
+		it('renderMessages should render errors inside a div having class alert-error', function () {
+			var $messages = $('div.alert-error').children('div');
+			expect($messages[0]).toContainText('Error two');
+			expect($messages[1]).toContainText('Error one');
+		});
+
+		it('removeMessages should remove all content from system-message-container', function () {
+			Joomla.removeMessages();
+			expect($("#system-message-container")).toBeEmpty();
+		});
+	});
+
+	describe('Core Joomla.isChecked', function () {
+		var form = document.getElementById('ischecked-test-form');
+		form.boxchecked = {value: 5};
+
+		beforeAll(function () {
+			Joomla.isChecked(true, form);
+		});
+
+		it('should increase form.boxchecked.value from 5 to 6', function () {
+			expect(form.boxchecked.value).toEqual(6);
+		});
+		it('should set checkAllToggle.checked to false', function () {
+			expect(form.elements[ 'checkall-toggle' ].checked).toEqual(false);
+		});
+	});
+
+	describe('Core Joomla.tableOrdering', function () {
+		beforeAll(function () {
```

Comment:
The indentation does not look right below here. Can you check @Ruchiranga ? :-) Thank you